client/state/analytics/test/middleware.js

```diff
@@ -19,153 +14,119 @@ import {
 	loadTrackingTool,
 } from '../actions';
 import { analyticsMiddleware } from '../middleware.js';
-import { spy as mockAdTracking } from 'calypso/lib/analytics/ad-tracking';
-import { spy as mockMC } from 'calypso/lib/analytics/mc';
-import { spy as mockGA } from 'calypso/lib/analytics/ga';
-import { spy as mockPageView } from 'calypso/lib/analytics/page-view';
-import { spy as mockTracks } from 'calypso/lib/analytics/tracks';
+import {
+	trackCustomFacebookConversionEvent,
+	trackCustomAdWordsRemarketingEvent,
+} from 'calypso/lib/analytics/ad-tracking';
+import { bumpStat as mcBumpStat } from 'calypso/lib/analytics/mc';
+import { gaRecordPageView, gaRecordEvent } from 'calypso/lib/analytics/ga';
+import { recordPageView as pageviewRecordPageView } from 'calypso/lib/analytics/page-view';
+import {
+	setTracksOptOut as tracksSetTracksOptOut,
+	recordTracksEvent as tracksRecordTracksEvent,
+} from 'calypso/lib/analytics/tracks';
 import { addHotJarScript } from 'calypso/lib/analytics/hotjar';
 
 const noop = () => {};
 
-jest.mock( 'calypso/lib/analytics/page-view', () => {
-	const pageViewSpy = require( 'sinon' ).spy();
-	const { pageViewMock } = require( './helpers/analytics-mock' );
-
-	const mock = pageViewMock( pageViewSpy );
-	mock.spy = pageViewSpy;
-
-	return mock;
-} );
-
-jest.mock( 'calypso/lib/analytics/tracks', () => {
-	const tracksSpy = require( 'sinon' ).spy();
-	const { tracksMock } = require( './helpers/analytics-mock' );
-
-	const mock = tracksMock( tracksSpy );
-	mock.spy = tracksSpy;
-
-	return mock;
-} );
-
-jest.mock( 'calypso/lib/analytics/ad-tracking', () => {
-	const adTrackingSpy = require( 'sinon' ).spy();
-	const { adTrackingMock } = require( './helpers/analytics-mock' );
-
-	const mock = adTrackingMock( adTrackingSpy );
-	mock.spy = adTrackingSpy;
-
-	return mock;
-} );
-
-jest.mock( 'calypso/lib/analytics/mc', () => {
-	const mcSpy = require( 'sinon' ).spy();
-	const { mcMock } = require( './helpers/analytics-mock' );
-
-	const mock = mcMock( mcSpy );
-	mock.spy = mcSpy;
+jest.mock( 'calypso/lib/analytics/page-view', () => ( {
```

Comment:
This file was using a weird mocking pattern, ultimately mocking these modules to a different API (a `spy` export) than the actual module (a `bumpStat` export, for example). That is confusing and it was making eslint's `import/named` unhappy.

I refactored this file to use Jest mocks instead of Sinon's, and changed the mocked object to maintain the original API.